Merge dev -> main

darkbringer1 opened the pull request.

darkbringer1 committed:
Add community logo asset

darkbringer1 committed:
Add new community logo assets

darkbringer1 committed:
Create README_en.md

First commit of readme file in english

darkbringer1 committed:
Update README_en.md

Add shortcuts to sections

darkbringer1 committed:
Create LICENSE

darkbringer1 committed:
Create LICENSE

darkbringer1 committed:
Create CODE_OF_CONDUCT.md

darkbringer1 committed:
Create PULL_REQUEST_TEMPLATE.md

darkbringer1 committed:
Rename README_en.md to README.md

darkbringer1 committed:
Fix broken target objects after updating tuist 4.6.0

darkbringer1 committed:
Update tuist dependency package file path

darkbringer1 committed:
Update tuist dependency graph for framework types

darkbringer1 committed:
Add .build file to .gitignore

berkaytncl committed:
Fix package dependencies

Remove:
-Framework dependencies
Add:
-Spm dependencies to the project
-Firebase auth initialization

# Conflicts:
#	Project.swift
#	Resources/GoogleService-Info.plist
#	Targets/SwiftBuddiesLogin/Sources/LoginView.swift
#	Targets/SwiftBuddiesMain/Sources/AppMain.swift
#	Tuist/Package.swift

berkaytncl committed:
Fix package dependencies

Remove:
-Framework dependencies
Add:
-Spm dependencies to the project
-Firebase auth initialization

# Conflicts:
#	Project.swift
#	Resources/GoogleService-Info.plist
#	Targets/SwiftBuddiesMain/Sources/AppMain.swift
#	Tuist/Package.swift

canyoldas0 committed:
fixed tuist build

canyoldas0 committed:
first commit

canyoldas0 committed:
fixed tuist config

canyoldas0 committed:
Merge remote-tracking branch 'origin/can/tuist-fix' into dev

darkbringer1 committed:
Remove .package.resolved file to avoid build tools conflicts

darkbringer1 committed:
Re-add readme.md file

darkbringer1 committed:
Add app icon

darkbringer1 committed:
Remove redundant app icon

baskurthalit committed:
Onboarding feature developed.

darkbringer1 committed:
Add Pull Request Template

baskurthalit committed:
Custom corner radius shape deleted.

baskurthalit committed:
Merge branch 'dev' into feature-Onboarding

baskurthalit committed:
Some logics for OnboardingView in main target are moved module itself.
LocalDataManager deleted.
IsSeenOnboarding is handled by @AppStorage()
Remove unnecessary lines.

baskurthalit committed:
Remove unnecessary default values.

baskurthalit committed:
Default func parameters updated.

darkbringer1 committed:
Update app tab view

baskurthalit committed:
Merge branch 'dev' into feature-Onboarding

# Conflicts:
#	SwiftBuddiesIOS/Sources/ContentView.swift

canyoldas0 committed:
Onboarding Feature (#17)

### Task Link :link:

[Task](https://github.com/orgs/SwiftBuddiesTR/projects/3?pane=issue&itemId=57480927)
<!-- What is your redmine link for this task? -->

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->
Develop OnboardingModule for the project.

### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->
Created a module called OnboardingModule.
Developed OnboardingView with two slides.

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->
Checked in different screen sizes.

### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->
<img
src="https://github.com/SwiftBuddiesTR/BuddiesIOS/assets/57860204/b2d65d6b-a09e-415c-a998-9afab05f76fe"
width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

berkaytncl committed:
Add GoogleService-Info.plist and assets

berkaytncl committed:
Add GoogleService-Info.plist and assets

berkaytncl committed:
Add login module and change dependencies

berkaytncl committed:
Add login module and change dependencies

berkaytncl committed:
Login feature developed

berkaytncl committed:
Login feature developed

berkaytncl committed:
Refactor code

berkaytncl committed:
Refactor code

berkaytncl committed:
Remove unused dependencies

berkaytncl committed:
Remove unused dependencies

berkaytncl committed:
Change TabView subviews

Change:
- About target to Profile target
Add:
- SettingsView and AboutView

berkaytncl committed:
Change TabView subviews

Change:
- About target to Profile target
Add:
- SettingsView and AboutView

berkaytncl committed:
Fix property and property name

berkaytncl committed:
Fix property and property name

darkbringer1 committed:
Update README.md

berkaytncl committed:
Remove unnecessary comment lines from the project

berkaytncl committed:
Remove unnecessary comment lines from the project

berkaytncl committed:
Move NavigationStack into ProfileView and Add NotificationCenter for logout functionality

berkaytncl committed:
Move NavigationStack into ProfileView and Add NotificationCenter for logout functionality

berkaytncl committed:
Improve UX with code enhancements

berkaytncl committed:
Improve UX with code enhancements

canyoldas0 committed:
added localization codegen

canyoldas0 committed:
Merge branch 'can/localization-codegen' into dev

berkaytncl committed:
Refactor AuthenticationManager according to code review

berkaytncl committed:
Refactor AuthenticationManager according to code review

oguzhanabuhanoglu committed:
LocationModel & MapViewModel added, map initialised w/ user location

oguzhanabuhanoglu committed:
NewEventView designed for create new event.

darkbringer1 committed:
Localizable generated

oguzhanabuhanoglu committed:
i just edited the file locations

oguzhanabuhanoglu committed:
add CoreData to save new events

oguzhanabuhanoglu committed:
CoreData cancelled

I will use swiftData

oguzhanabuhanoglu committed:
EventModel created for SwiftData

oguzhanabuhanoglu committed:
SwiftData added to save new events

oguzhanabuhanoglu committed:
query for events

oguzhanabuhanoglu committed:
Create map w/UIViewRepresentable to get coordinate w/longPressGestureRecognizer

oguzhanabuhanoglu committed:
Save location latitude and longitude, move them to mapView

oguzhanabuhanoglu committed:
custom annotation view created, put it on map view for events

oguzhanabuhanoglu committed:
Requesting swift data has been moved to mapView

oguzhanabuhanoglu committed:
Use navigationStack instead of navigationView

oguzhanabuhanoglu committed:
Different pins for all categories, show all pins at beginning

oguzhanabuhanoglu committed:
changing some file names also add another viewModel

oguzhanabuhanoglu committed:
Changed category property in Event model to enum and added "All" filter

oguzhanabuhanoglu committed:
edit some codes

oguzhanabuhanoglu committed:
EventListView Created and UI Updated

oguzhanabuhanoglu committed:
EventListViewCreated and UI updated

oguzhanabuhanoglu committed:
EventDetailsView Created

oguzhanabuhanoglu committed:
pr

berkaytncl committed:
Merge branch 'login-feature' of https://github.com/SwiftBuddiesTR/BuddiesIOS into login-feature

# Conflicts:
#	Project.swift
#	SwiftBuddiesIOS/Targets/LocalizationModule/Resources/Localizable.xcstrings

berkaytncl committed:
Confict resolved

berkaytncl committed:
Add Network module, Remove email and anonymously login

berkaytncl committed:
Remove Firebase from project, Refactor Network module

oguzhanabuhanoglu committed:
Add alert pop-up's

myildirim48 committed:
Update ReadME logo url

oguzhanabuhanoglu committed:
fix

oguzhanabuhanoglu committed:
date formatter

darkbringer1 committed:
remove event singleton and pass newEventModel to locationSeletionView

oguzhanabuhanoglu committed:
add plist req string

oguzhanabuhanoglu committed:
LocationManager added, category enum changed to stringArray

darkbringer1 committed:
Update projet file

darkbringer1 committed:
Initialize BuddiesNetwork

darkbringer1 committed:
Add backend request after signin/register

oguzhanabuhanoglu committed:
The locationManager and mapViewModel have been updated.

oguzhanabuhanoglu committed:
cleaner code.

oguzhanabuhanoglu committed:
add color extension and clenar code

oguzhanabuhanoglu committed:
add displayFormatter to show dates on view.

oguzhanabuhanoglu committed:
LocationManager last update.

oguzhanabuhanoglu committed:
event category changed to Category model.

oguzhanabuhanoglu committed:
Change AnnotationView color and listRowView actón/

darkbringer1 committed:
Fix access token getter

Add
- Keychain manager

darkbringer1 committed:
Add Buddies auth service

darkbringer1 committed:
Add logout logic

canyoldas0 committed:
wip

KateKashko committed:
Rebased feed branch

darkbringer1 committed:
WIP: can

canyoldas0 committed:
fixed missing auth dependency for core module

oguzhanabuhanoglu committed:
Color issues solved, Deprecated map could not change.

oguzhanabuhanoglu committed:
SearchBar added on LocationSelectionView.

darkbringer1 committed:
Fix syntax

darkbringer1 committed:
Login feature (#21)

### Issue Link :link:
<!-- What is your redmine link for this task? -->

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->

**The primary goals of this pull request:**
1. Improve the login feature in the community app, working alongside
Oğuzhan.
2. Use MVVM architecture and SwiftUI for development.
3. Add Firebase for various login options like email, anonymous, Apple,
and Google login.

### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->

We've enhanced the login functionality of the community app, choosing
MVVM for its organization and SwiftUI for its simplicity. Firebase
integration adds versatile login methods, improving user experience and
security.

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->

**TBD** 
### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->

<img
src="https://github.com/SwiftBuddiesTR/BuddiesIOS/assets/98055405/6a4939c2-1f11-4ad4-bb8d-0b4348861ad8.gif"
width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

fatmanurshain reviewed (dismissed):
'APPROVE' performed by Xcode.

oguzhanabuhanoglu committed:
create endpoinst to connect with backend

KateKashko committed:
add feed network

oguzhanabuhanoglu committed:
add location permission into tuist.

darkbringer1 committed:
Fix login

Fix an issue where user forced to login again after app restart

darkbringer1 committed:
Fix login (#32)

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->
- Fix an issue where user forced to login again after app restart

### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->
- Add a token check logic to buddies auth service

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->
- Close and open app and see if user logged out 

### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->

<img
src="https://github.com/user-attachments/assets/31fc37c6-d95e-4b5b-a83e-c7a847e2748a"
width="250">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

oguzhanabuhanoglu committed:
WIP: fix

oguzhanabuhanoglu committed:
Database Manager Created, backend integrated without swift data.

oguzhanabuhanoglu committed:
update responnse model

darkbringer1 committed:
Update map region set

Change model property

darkbringer1 committed:
update new event view ui

darkbringer1 committed:
update map view components

darkbringer1 committed:
add contributors flow

darkbringer1 committed:
Fix endpoint issues and errors

darkbringer1 committed:
WIP update models and add filter view

darkbringer1 committed:
Update github events

darkbringer1 committed:
Add a JsonDecodingInterceptor to check and print which fields are getting decode error

darkbringer1 committed:
update design

darkbringer1 committed:
add pagination

darkbringer1 committed:
update wip

darkbringer1 committed:
update contributer detail ux

darkbringer1 committed:
Remove feed

darkbringer1 committed:
Add mock feed views and flows

darkbringer1 committed:
Map view (#25)

### Issue Link :link:
<!-- What is your redmine link for this task? -->
[Locations Task](https://github.com/SwiftBuddiesTR/BuddiesIOS/issues/12)

### Goals :soccer:
- Map üzerinde etkinlik oluşturmak için Plus (+) butonu eklendi.

- Kullanıcı, etkinlik parametrelerini girebileceği bir UI'ye
yönlendirildi.

- Tüm parametreler girildikten sonra, longPressGesture ile etkinlik
konumu seçilebilecek bir UI sağlandı.

- Etkinlik oluşturulduktan sonra, MainMapView üzerinde diğer
etkinliklerle birlikte görüntülendi.

- MainMapView üzerinde etkinlikler arasında smooth animasyonlarla geçiş
yapılabilir.

- Kırmızı info butonu ile etkinlik detaylarının görüntülendiği bir UI'ye
erişim sağlandı.

- Etkinlikler, kategorilerine göre filtrelenebilir ve aranan etkinlikler
kolayca bulunabilir.


### Implementation Details :construction:
- MVVM Mimarisi kullanıldı.

- NetworkManager ile iletişim yine local managerlar üzerinden kurularak
istekler atıldı.

- LongPressGesture tam verimle çalışması için ikinci map ekranı için
RepresentableMapView yazıldı.

- Custom anotastonlar ve viewlar kullanmaya özen gösterildi.

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->

### Screenshots/Gifs :camera:
<div align="center">

<!-- İlk Satır -->
<img
src="https://github.com/user-attachments/assets/c55444dc-f306-43d3-8baf-7c30541d2588"
width="150">
<img
src="https://github.com/user-attachments/assets/788169c8-082a-4534-a670-47fcd0297128"
width="150">
<img
src="https://github.com/user-attachments/assets/6cd40ce7-ac85-42f1-9456-6aa9319e2c81"
width="150">

<!-- Boşluk -->
<br>

<!-- İkinci Satır -->
<img
src="https://github.com/user-attachments/assets/0a026bc3-527e-4558-9c32-887a81b03ab4"
width="150">
<img
src="https://github.com/user-attachments/assets/45641b07-eb2e-4165-84db-d6741ed5e005"
width="150">
<img
src="https://github.com/user-attachments/assets/5da18bef-a5a8-4a0e-9dec-fb1a63b88065"
width="150">



</div>



## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

darkbringer1 committed:
add cache

darkbringer1 committed:
add cache

darkbringer1 committed:
update contributors details

canyoldas0 committed:
Contributors (#33)

### Issue Link :link:
<!-- What is your redmine link for this task? -->

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->

### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->

### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->
How to add gif/screenshots: 
1. Drag and drop the gif/screenshot here, let it upload.
2. Copy the link and paste it like below ->
```
<img src="LINK-HERE.gif" width="150">
```
3. add the appropriate file extension after it "*.gif" or "*.jpg"
- Important: Be careful not to exceed 150 points width that is
specified. It makes it harder chech overall ui elements if width is
wider than 150 points.

Example usage: 

<img
src="https://github.com/SwiftBuddiesTR/BuddiesIOS/assets/40922278/292b39cd-898a-4233-ad6f-137b28351964.gif"
width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

darkbringer1 committed:
Resturcture files and fix endpoints

darkbringer1 committed:
Implement add post view and network

darkbringer1 committed:
add pagination to feed

darkbringer1 committed:
update addpost view

darkbringer1 committed:
add photo upload encoding and decoding from base64

darkbringer1 committed:
update base64 decode encode methods

darkbringer1 committed:
update proj

darkbringer1 committed:
Fix apple sign in

Update proejct entitlements

berkulucay committed:
Add dark&tinted app logos

berkulucay committed:
Update: minor changes to app logos

darkbringer1 committed:
Can berk/logo update (#34)

darkbringer1 committed:
Update image upload strategy

darkbringer1 committed:
Merge branch 'feed-cell-ui' into dev

Fatihozn committed:
create profile module

darkbringer1 committed:
Update README.md file image resource url (#26)

### Issue Link :link:
https://github.com/SwiftBuddiesTR/BuddiesIOS/issues/23

### Goals :soccer:
To show the community logo in the README file

### Implementation Details :construction:
There is no architectural changes or new functionality.

### Testing Details :mag:
Reviewed Changes in Pushed Branch

### Screenshots/Gifs :camera:
<img width="875" alt="Screenshot 2024-09-11 at 02 30 18"
src="https://github.com/user-attachments/assets/f24721d8-852d-42bb-933d-79cb820330ab">


## PR Type
What kind of change does this PR introduce?
- [x] Documentation content changes

Fatihozn committed:
UI of profile and edit page has done

oguzhanabuhanoglu committed:
update userLocation and fetch eventData

Fatihozn committed:
profile and edit API Calls complete

oguzhanabuhanoglu committed:
code refactor, change annotation colors and debug selectedItem issue

oguzhanabuhanoglu committed:
"Buddy"

darkbringer1 committed:
Update userLocation and fetch eventData (#39)

### Issue Link :link:
<!-- What is your redmine link for this task? -->
[37](https://github.com/SwiftBuddiesTR/BuddiesIOS/issues/37)
[38](https://github.com/SwiftBuddiesTR/BuddiesIOS/issues/38)

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->
Update userLocation and fetch eventData

### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->
I resolved the user location update issue and optimized the GET request
with a caching mechanism which is new in network layer.

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->

### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->

<img
src="https://github.com/user-attachments/assets/78b1edcb-d2a0-41a1-aebe-1f520ac88663"
width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

-[x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

darkbringer1 committed:
Profile (#40)

### Issue Link :link:
<!-- What is your redmine link for this task? -->

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->
Profile and profile edit page created
### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->
Profile module (Profile page and profile edit page) created including
API Calls
### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->

### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->
<img
src="https://github.com/user-attachments/assets/ecbcd8ea-e2aa-4be5-ac44-7a70e4669b94"
width="150">
<img
src="https://github.com/user-attachments/assets/c4cb6ea7-c814-4a00-b7bd-a15d5dba22f9"
width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

darkbringer1 committed:
Fix image get method to accommodate v2 api version

darkbringer1 committed:
update minimum deployment target

darkbringer1 committed:
Fix feed and contributors view factory methods to prevent unnecessary deallocation of view in certain cases

darkbringer1 committed:
Refactor feed view

darkbringer1 committed:
Export feed post view to DesignModule

darkbringer1 committed:
fix fetch logic

darkbringer1 committed:
Feed cell UI (#42)

Update feed views

darkbringer1 committed:
update profile views

darkbringer1 committed:
add new color palette

darkbringer1 committed:
WIP update button styles

darkbringer1 committed:
add colors

darkbringer1 committed:
WIP started localizing project

darkbringer1 committed:
Update button showcase

GroophyLifefor committed:
Readme logo path fix

darkbringer1 committed:
Readme logo path fix (#45)

canyoldas0 committed:
update profile views (#43)

### Issue Link :link:
<!-- What is your redmine link for this task? -->

### Goals :soccer:
<!-- List the high-level objectives of this pull request. -->
<!-- Include any relevant context. -->
- Update profile view to comply with app architecture
### Implementation Details :construction:
<!-- Explain the reasoning behind any architectural changes. -->
<!-- Highlight any new functionality. -->
- Implement navigation stack
- Rearrange filesystem
- Create relevant view files
- Declutter views
- Extract view components
- Add button showcase

### Testing Details :mag:
<!-- Describe what tests you've added for your changes. -->
-
### Screenshots/Gifs :camera:
<!-- Add screenshots or Gifs when appropriate. -->

<img
src="https://github.com/user-attachments/assets/8d0ab1f3-2ef1-41f7-89be-21271baa87e9"
width="150">

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

aniltaskiran committed:
move network module to target

aniltaskiran committed:
refactor on tuist

darkbringer1 committed:
Refactor on tuist (#46)

## Summary

This MR implements a modular architecture for the BuddiesIOS app using
Tuist for project generation. The modular approach divides the
application into separate concerns, enhancing maintainability,
testability, and collaboration efficiency among team members. Each
module is structured to be self-contained with clear dependencies,
promoting clean architecture principles.

## Relevant domains
- Architecture
- Project Structure
- Dependency Management

## Structure

The modular architecture is implemented using the following key
components:

- Created modular framework structure with the following modules:
  - `Auth`: Authentication functionality
  - `Contributors`: GitHub contributors display
  - `Core`: Core application features and dependency container
  - `Design`: UI components and design system
  - `Feed`: Social feed functionality
  - `Localization`: Localization resources
  - `Network`: Networking layer

Dependencies between modules are clearly defined, with each module
specifying its dependencies in the `Project.swift` file. The
architecture follows a clean dependency graph to prevent circular
dependencies.

```mermaid
graph TD
    A[App] --> B[Core]
    A --> C[Design]
    A --> D[Auth]
    A --> E[Contributors]
    A --> F[Feed]
    B --> D
    B --> G[Network]
    C --> H[Localization]
    E --> B
    E --> C
    F --> B
    F --> C
    G --> I[BuddiesNetwork]
    D --> G
```

Each module is configured using the `Project.module` extension that
standardizes the module creation process, including:
- Consistent naming conventions
- Common deployment targets
- Resource management
- Module dependencies

darkbringer1 committed:
graph

darkbringer1 committed:
feat: add present page logic to coordinator
